napari_allencell_annotator/widgets/files_widget.py

```diff
@@ -55,6 +55,29 @@ def __init__(self, annotator_model: AnnotatorModel):
             lambda: self._handle_shuffle(self._annotator_model.is_images_shuffled())
         )
         self._annotator_model.image_count_changed.connect(self._handle_image_count_change)
+        self.currentItemChanged.connect(self._handle_item_changed)
+
+    def _handle_item_changed(self, curr_item: FileItem, prev_item: FileItem) -> None:
+        """
+        Update the model when the current item in the file widget is changed if it has not been updated.
+
+        Parameters
+        ----------
+        curr_item:
+            the current file item
+        prev_item:
+            the previous file item
+        """
+        if prev_item is None:
+            self._annotator_model.set_previous_image_index(None)
```

Comment:
leave comment why we need to re-set this to none or remove if not needed

Reply:
Model should change this only once ever
1) AnnotatorView calls model function `next_image()` which emits a next_image signal
2) FileWidget subscribes to `next_image` signal and sets current item to next item
3) model updates in `_handle_item_changed()`

same for previous

Then we can refactor this handle item changed function to just update the model

Reply:
When we clicks on the first image to appear, prev_item here is None. If we don't include this line, it will call self._annotator_model.set_previous_image_index(self.row(None)), which set the previous image index to -1 in the model. This causes an error when we want to unhighlight the previous item in the function _display_img in images_view.py.

Reply:
Similarly, we need this for set_curr_img_index as it will cause the same problem when we clear the display after shuffling and deleting images (assigning the current index to None). It will try to highlight the next item label, which doesn't exist.